sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -542,5 +540,18 @@ public Mono<String> getBodyAsString() {
         public Mono<String> getBodyAsString(Charset charset) {
             return getBodyAsByteArray().map(bytes -> new String(bytes, charset));
         }
+
+        @Override
+        public BinaryData getBodyAsBinaryData() {
+            BinaryData content = actualResponse.getBodyAsBinaryData();
+            doLog(content.toString());
```

Comment:
Why is this API override logging?

Reply:
To make body logging synchronous: `getBodyAsBinaryData` is the only sync way to get the response body and default implementation is async

https://github.com/Azure/azure-sdk-for-java/blob/1cb5a6a84f86f7cce5f0fa287b3980c5c4507ab3/sdk/core/azure-core/src/main/java/com/azure/core/http/HttpResponse.java#L89-L92

Reply:
Should we eagerly call `content.toString()` here? Instead we should consider passing the `content` to doLog() method and there, we can use the supplier methods ClientLogger provides to log if logging is enabled - `logBuilder.addKeyValue(BODY_KEY, () -> content.toString())`.

Reply:
`LoggingHttpResponse` is created only if logging at that level is enabled, body logging is enabled and content type is not a stream, so there is no need to do lazy `toString` - there are no checks in `doLog` method.

https://github.com/Azure/azure-sdk-for-java/blob/41038467140a89185d27ee9a0857a59acad0ef61/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java#L315